src/views/fileDecorations.ts

```diff
@@ -3,48 +3,27 @@ import * as vscode from "vscode";
 import { Event } from "vscode";
 
 import { getFromCache } from "../utils/addonCache";
-import { getRootFolderPath } from "../utils/reviewRootDir";
+import { splitUri } from "../utils/splitUri";
 
 export async function fileHasComment(uri: vscode.Uri) {
-  const fullpath = uri.fsPath;
-  const rootFolder = await getRootFolderPath();
-
-  const filepath = fullpath.replace(rootFolder, "");
-  const splitPath = filepath.split("/");
-  const guid = splitPath[1];
-  const keys = splitPath.slice(2);
-
-  const comments = await getFromCache(guid, keys);
-  // check if comments map is empty or not defined
-  if (!comments || Object.keys(comments).length === 0) {
-    return false;
-  }
-  return true;
-}
-
-export async function folderHasComment(uri: vscode.Uri) {
-  const fullpath = uri.fsPath;
-  const rootFolder = await getRootFolderPath();
-
-  const filepath = fullpath.replace(rootFolder, "");
-  const splitPath = filepath.split("/");
-  const guid = splitPath[1];
-  const keys = splitPath.slice(2);
-
-  // if there no keys, then we are at the guid path. return false
-  if (keys.length === 0) {
-    return false;
-  }
-
-  const comments = await getFromCache(guid, keys);
-  // check if comments map is empty or not defined
-  if (!comments || Object.keys(comments).length === 0) {
-    return false;
+  const { guid } = await splitUri(uri);
+  const comments = await getFromCache(guid, ["comments"]);
+
+  for (const version in comments) {
+    for (const filepath in comments[version]) {
+      for (const lineNumber in comments[version][filepath]) {
+        const { uri: currUri } = comments[version][filepath][lineNumber];
+        if (uri.fsPath === currUri.fsPath) {
+          return true;
+        }
+      }
+    }
```

Comment:
This logic can be simplified by statically accessing decedents of the `comments` object. The `splitUri()` function returns an object that gives us most of what we need to replicate what the current set of nested `for` loops are doing.

```suggestion
  const { guid, version, filePath } = await splitUri(uri);
  const comments = await getFromCache(guid, ["comments"]);

  if (comments?.[version]?.[filePath]) {
    return true;
```

In the code above we use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) is used here to ensure we don't throw if no relevant comments exist.  

This approach flips the perspective of what the function is checking. Rather looping through every comment and trying to match the current file's URI against the comment URI, we check if the comments object has an entry for this specific file. The success or failure of this approach depends on the correctness of our data model; if we retain unnecessary file objects or use these objects to store data other than comments the assumptions made by this approach will not hold.